libs/designer-v2/src/lib/core/actions/bjsworkflow/add.ts

```diff
@@ -482,10 +483,32 @@ export const trySetDefaultConnectionForNode = async (
   nodeId: string,
   connector: Connector,
   dispatch: AppDispatch,
-  isConnectionRequired: boolean
+  isConnectionRequired: boolean,
+  getState: () => unknown
 ) => {
   const connectorId = connector.id;
-  const connections = (await getConnectionsForConnector(connectorId)).filter((c) => c.properties.overallStatus !== 'Error');
+  let connections = (await getConnectionsForConnector(connectorId)).filter((c) => c.properties.overallStatus !== 'Error');
+
+  // Filter out dynamic connections unless we're in an A2A workflow AND inside an agent loop
+  // Only apply this filter if there are connections with the 'feature' property set
+  // Filtering behavior:
+  //   - Conversational agent (isA2A=true) + inside loop (isAgentSubgraph=true) → Allow dynamic connections
+  //   - Conversational agent (isA2A=true) + outside loop (isAgentSubgraph=false) → Filter out dynamic connections
+  //   - Autonomous agent (isA2A=false) + inside loop (isAgentSubgraph=true) → Filter out dynamic connections
+  //   - Autonomous agent (isA2A=false) + outside loop (isAgentSubgraph=false) → Filter out dynamic connections
+  //   - Regular workflow (isA2A=false) → Filter out dynamic connections
+  const hasDynamicConnections = connections.some((c) => c.properties.features || c.properties.feature);
+  if (hasDynamicConnections) {
+    const state = (getState() as RootState).workflow;
+    const isA2A = isA2AWorkflow(state);
+    const isAgentSubgraph = nodeId ? checkIsAgentSubgraph(nodeId, state.nodesMetadata) : false;
+
+    if (!isA2A || !isAgentSubgraph) {
+      // Filter out dynamic connections (check both 'features' and 'feature' for compatibility)
+      connections = connections.filter((c) => !equals(c.properties.features ?? c.properties.feature ?? '', 'DynamicUserInvoked', true));
```

Comment:
can we do this the opposite way?

You don't need hasDynamicConnections, just whatever in that block can be used directly